### PR TITLE
VIA compile fixes

### PR DESCRIPTION
**Pull request by Xelus22**
## Description

Various VIA bug compile fixes.

https://github.com/Xelus22/QMK-VIA-Hex/runs/2709257229?check_suite_focus=true

## Types of Changes

- [ ] Core
- [x] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* Caps Unlocked CU80 V2 ANSI RGB `LTO_ENABLE = yes`
* Caps Unlocked CU80 V2 ISO RGB `LTO_ENABLE = yes`
* mt64rgb apply #12697 also to the VIA keymap
* nopunin10did/jabberwocky fix correct layout
* rubi fix encoder 

## Checklist

- [ ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ ] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by zvecr — APPROVED** [review]

File: keyboards/rubi/rubi.c [hunk 1]
```diff
@@ -101,8 +101,9 @@ __attribute__ ((weak)) void encoder_update_user(uint8_t index, bool clockwise) {
...
         }
     }
+    return true;
 }
 
-void encoder_update_kb(uint8_t index, bool clockwise) {
-    encoder_update_user(index, clockwise);
+bool encoder_update_kb(uint8_t index, bool clockwise) {
+    return encoder_update_user(index, clockwise);
 }
```
Comment (tzarc) [inline keyboards/rubi/rubi.c:109]: Shall we just remove this, given that it's the default?

Comment (drashna) [inline keyboards/rubi/rubi.c:109]:
```suggestion
```

File: keyboards/rubi/rubi.c [hunk 2]
```diff
@@ -73,7 +73,7 @@ bool led_update_kb(led_t led_state) {
     return true;
 }
 
-__attribute__ ((weak)) void encoder_update_user(uint8_t index, bool clockwise) {
+__attribute__ ((weak)) bool encoder_update_user(uint8_t index, bool clockwise) {
```
Comment (drashna) [inline keyboards/rubi/rubi.c:76]:
```suggestion
bool encoder_update_kb(uint8_t index, bool clockwise) {
    if (!encoder_update_user(index, clockwise)) { return false; }
```

File: keyboards/rubi/rubi.c [hunk 3]
```diff
@@ -73,7 +73,7 @@ bool led_update_kb(led_t led_state) {
...
     if (index == 0) {
```
Comment (tzarc) [inline keyboards/rubi/rubi.c:77]: Should probably be in the keymap.

**Review by tzarc — CHANGES_REQUESTED** [review]

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/rubi/rubi.c (hunk 2 above)
Comment (tzarc) [reply keyboards/rubi/rubi.c:76]: Probably better to move the `_user` to the keymap and leave it out of the keyboard entirely.

Comment (Xelus22) [reply keyboards/rubi/rubi.c:76]: Moving to user keymap.

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Pull request merged**